protos/core/Btfs.proto

```diff
@@ -0,0 +1,256 @@
+    	CANCELED = 4; //reserve state, for some situation that the subcontract was closed abnormally
+	}
+	State state = 18;
+	bytes escrow_signature = 19; //signature got from escrow while last payment status check
+	Timestamp status_time = 20;  //current status generated time
+	bytes guard_signature = 21; //guard signature for previous information
+}
+
+message ContractStatus{
+	bytes renter_address = 1; // buyer wallet address
+	bytes guard_address =2; //guard wallet address
+	bytes escrow_address = 3; //escrow wallet address
+	bytes file_hash = 4; //file's hash 
+	int64 file_size = 5; //file's size
+	Timestamp rent_start = 6;
+	Timestamp rent_end = 7;
+	int chunk_count = 8; //chunk's count
+	int recover_threshold = 9; //reserved field, indicate the right chunk count for the file
+	int check_frequence = 10; //check's frequency, 1 means check once per year
+	repeated SubContractStatus SubContractStatus = 11; //all sub contract's status
+	Timestamp status_time = 12;  //current status generated time
+	bytes previous_status_id = 13; //previous contract status id, if it is first, mark it as 0
+	bytes status_id = 14; //the hash of the current status
+	bytes signature = 15; //guard's signature of such status
+}
+
+message UploadStatusReport{
+	bytes renter_address = 1; 
+	bytes host_address = 2;
+	bytes chunk_hash = 3;
+	int64 chunk_size = 4;
+	int64 upload_size = 5;
+	int64 bandwidth_since_last_report = 6; //bps since last bandwidth_since_last_report
+	int64 connection_lost_times = 7;
+	Timestamp report_time = 8;
+	bytes signature = 9; //renter's signature of the report
+}
+
+message CheckContractRequest{
+	bytes file_hash = 1;
+	bytes renter_address = 2;
+	bytes requester_address = 3;
+	bytes signature =4; //requester signed 
+}
+
+message StartEscrowRequest{
+	ProofOfFund escrow_amount = 1; // the proof of renter had reserve money in the escrow for such chunk 
+	ProofOfFund escrow_collateral_amount = 2; //the proof of host for collateral
+	int pay_frequency = 3; 
+	Timestamp start_time = 4;
+	Timestamp end_time = 5;
+	bytes peerId = 6; //guard's instance's peerId
+	bytes signature = 7; 
+}
+
+message PaymentStatus{ 
+	//fields need by guard for escrow payment's staus, need ESCROW decision and revise
+	bytes from_address = 1; //the fund's original address, in BTFS it should be renter_address
+	bytes to_address = 2; //the fund's target wallet address, in BTFS it should be host's address
+	bytes manager_address = 3; //the manager's wallet address, in BTFS it should be guard's address
+	enum status{
+		RUNNING = 1;
+		PAUSE = 2;
+		DONE = 3;
+	}
+	status status = 4;
+	bytes payment_id = 5; //escrow assigned id for the payment
+	int64 total_amount = 6; //whole amount of the subcontract , equal rent amount + collateral amount
+	int64 collateral_amount = 7;
+	int64 withhold_amount = 8;
+	int64 paid_amount = 9;
+	int64 next_payment_amount = 10;
+	Timestamp next_payment_time = 11;
+	Timestamp last_payment_time = 12;
+	Timestamp sign_time = 13;
```

Comment:
These info btfs guard already have, better don't need escrow to return when create escrow.